api/src/main/java/jakarta/data/metamodel/ComparableAttribute.java

```diff
@@ -109,22 +78,3 @@ static <T, V extends Comparable<V>> ComparableAttribute<T, V> of(
     }
 }
 
-/**
- * Hidden internal implementation of ComparableAttribute.
- *
- * @param <T> entity class of the static metamodel.
- * @param <V> type of entity attribute (or wrapper type if primitive).
- */
-record ComparableAttributeRecord<T,V extends Comparable<V>>(String name)
```

Comment:
This PR mostly does well with keeping the introduced implementation classes/records as package protected and hidden, except not the *AttributeRecord classes.
I'm very much opposed to introducing more implementation classes into the API.  This class here was demonstrating one way you can still have implementation for Attribute classes and avoid including it in the API.  I suppose it's fine if we're going to temporarily do this and put it back before releasing, but I want to be sure that we actually do that.